src/PointwiseFunctions/AnalyticSolutions/NewtonianEulerSolutions/IsentropicVortex.hpp

```diff
@@ -0,0 +1,175 @@
+ * polytropic relation
+ *
+ * \f{align*}
+ * p = \rho^\gamma.
+ * \f}
+ *
+ * On the other hand, if the velocity along the \f$z-\f$axis is not a constant
+ * but a function of the \f$z\f$ coordinate, the resulting modified isentropic
+ * vortex is still a solution to the Newtonian Euler system, but with source
+ * terms that are proportional to \f$dv_z/dz\f$ (See
+ * NewtonianEuler::Sources::IsentropicVortexSource.)
+ *
+ * [1] H. Yee, N. Sandham, M. Djomehri, Low-dissipative high-order
+ * shock-capturing methods using characteristic-based filters, J. Comput. Phys.
+ * 150 (1999) 199, http://dx.doi.org/10.1006/jcph.1998.6177
+ */
+class IsentropicVortex {
+ public:
+  struct AdiabaticIndex {
+    using type = double;
+    static constexpr OptionString help = {"The adiabatic index of the fluid."};
+    // Note: bounds only valid for an ideal gas.
+    static type lower_bound() { return 1.0; }
+    static type upper_bound() { return 2.0; }
+    static type default_value() { return 5.0 / 3.0; }
+  };
+
+  struct Center {
+    using type = std::array<double, 3>;
+    static constexpr OptionString help = {
+        "The coordinates of the center of the vortex at t = 0."};
+    static type default_value() { return {{0.0, 0.0, 0.0}}; }
+  };
+
+  struct MeanVelocity {
+    using type = std::array<double, 3>;
+    static constexpr OptionString help = {
+        "The velocity of the mean fluid flow."};
+    static type default_value() { return {{1.0, 1.0, 1.0}}; }
+  };
+
+  struct Strength {
+    using type = double;
+    static constexpr OptionString help = {"The strength of the vortex."};
+    static type default_value() { return 3.0; }
+    static type lower_bound() { return 0.0; }
+  };
+
+  using options = tmpl::list<AdiabaticIndex, Center, MeanVelocity, Strength>;
+  static constexpr OptionString help = {"Newtonian Isentropic Vortex."};
+
+  IsentropicVortex() = default;
+  IsentropicVortex(const IsentropicVortex& /*rhs*/) = default;
+  IsentropicVortex& operator=(const IsentropicVortex& /*rhs*/) = default;
+  IsentropicVortex(IsentropicVortex&& /*rhs*/) noexcept = default;
+  IsentropicVortex& operator=(IsentropicVortex&& /*rhs*/) noexcept = default;
+  ~IsentropicVortex() = default;
+
+  IsentropicVortex(double adiabatic_index, Center::type center,
+                   MeanVelocity::type mean_velocity, double strength,
+                   const OptionContext& context = {});
+
+  explicit IsentropicVortex(CkMigrateMessage* /*unused*/) noexcept {}
+
+  template <typename DataType>
+  using tags = tmpl::list<MassDensity<DataType>, Velocity<DataType, 3>,
+                          SpecificInternalEnergy<DataType>>;
+
+  template <typename DataType>
+  tuples::TaggedTupleTypelist<tags<DataType>> variables(
+      const tnsr::I<DataType, 3>& x, const double& t) const noexcept;
+
+  // clang-tidy: no runtime references
+  void pup(PUP::er& /*p*/) noexcept;  //  NOLINT
+
+  inline constexpr double adiabatic_index() const noexcept {
```

Comment:
don't need to mark these inline (as they are defined within the class definition)